### PR TITLE
try adding a simple check for numeric

### PR DESCRIPTION
**Pull request by aaronjorbin**
Trac ticket: https://core.trac.wordpress.org/ticket/45018

### CODE REVIEW

**Comment by desrosj** [comment]
I updated the conditional to use `is_string()` instead of `false === is_numeric()`, it felt a little more readable to me. Those are the only two options for array indexes, so the `is_numeric()` in the `elseif` should not be needed.

There was another conditional a bit deeper that could also potentially cause this error when checking against old menu locations. There is no test for this scenario currently, though. The last test method can be copied and modified.

Also, I moved the `$limits` assignment closer to where the other value assignments are for more clarity.

@aaronjorbin is the first diff in `src/wp-admin/includes/ms.php` intentional?

File: src/wp-admin/includes/ms.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ function wpmu_delete_blog( $blog_id, $drop = false ) {
 	}
 
 	// Don't destroy the initial, main, or root blog.
-	if ( $drop && ( 1 == $blog_id || is_main_site( $blog_id ) || ( $blog->path == $current_network->path && $blog->domain == $current_network->domain ) ) ) {
+	if ( $drop && $blog && ( 1 == $blog_id || is_main_site( $blog_id ) || ( $blog->path == $current_network->path && $blog->domain == $current_network->domain ) ) ) {
```
Comment (pento) [inline src/wp-admin/includes/ms.php]: Why isn't this being caught in the `if ( $drop && ! $blog )` check just before it?

Comment (aaronjorbin) [reply src/wp-admin/includes/ms.php]: Hmm, that's a good question. The issue this was solving was that $blog wasn't an object. 

See: https://travis-ci.org/WordPress/wordpress-develop/jobs/445966625#L1411

I think that there is a chance we aren't setting the environment up completely correctly and that could be causing this.

File: src/wp-includes/nav-menu.php [hunk 2]
```diff
@@ -1139,7 +1139,10 @@ function wp_map_nav_menu_locations( $new_nav_menu_locations, $old_nav_menu_locat
 			foreach ( $registered_nav_menus as $new_location => $name ) {
 
 				// ...actually match!
-				if ( false === stripos( $new_location, $slug ) && false === stripos( $slug, $new_location ) ) {
+				if ( is_string( $new_location ) && false === stripos( $new_location, $slug ) && false === stripos(
```
Comment (aaronjorbin) [inline src/wp-includes/nav-menu.php]: This broke the tests.

**Comment by ocean90** [comment]
This was committed in https://core.trac.wordpress.org/changeset/43899.

**Pull request closed without merging**